File: client-leger/lib/utils/constants/tile-info/tile-info.dart
import 'package:client_leger/interfaces/game-tile.dart';
import 'package:client_leger/utils/enums/tile-type.dart';

List<GameTile> gameTiles = [
  GameTile(
    id: TileType.Ground.value,
    name: 'Gazon',
    image: 'assets/images/tiles/grass.jpg',
    descriptions: [
      "Tuile par défaut du jeu (tuile de terrain)",
      "Les joueurs et les objects peuvent y être posés dessus', 'coût: 1",
    ],
  ),

  GameTile(
    id: TileType.Ice.value,
    name: 'Glace',
    image: 'assets/images/tiles/ice.jpg',
    descriptions: [
      'Un joueur qui y marche dessus à 10% de chance de perdre pied et tomber, terminant instantanément le tour du joueur',
      'tant que le joueur se trouve sur de la glace, ses attributs « attaque » et « défense » souffrent d’un malus de 2.',
      'coût: 0',
    ],
  ),

  GameTile(
    id: TileType.Water.value,
    name: 'Eau',
    image: 'assets/images/tiles/water.jpg',
    descriptions: ['Tuile de terrain', 'Coût: 2'],
  ),
  GameTile(
    id: TileType.Wall.value,
    name: 'Mur',
    image: 'assets/images/tiles/wall.jpg',
    descriptions: [
      'Obstacle infranchissable à moins d’avoir un item spécial',
      'Aucun objet ne peut y être placé',
      'Pas considérée comme une tuile de terrain',
    ],
  ),
  GameTile(
    id: TileType.ClosedDoor.value,
    name: 'Porte fermée',
    image: 'assets/images/tiles/closed-door.jpg',
    descriptions: [
      "Une porte fermée doit être ouverte par le joueur avec le bouton 'Porte'.",
      "Sinon elle agit comme un mur infranchissable à moins d’avoir un item spécial",
    ],
  ),
  GameTile(
    id: TileType.OpenDoor.value,
    name: 'Porte ouverte',
    image: 'assets/images/tiles/open-door.jpg',
    descriptions: [
      "Une porte ouverte laisse passer le joueur.",
      "Sinon elle agit comme un mur infranchissable à moins d’avoir un item spécial",
    ],
  ),
];

File: common/tile-info.ts
import { GameTile } from './interfaces/game-tile'
import { TileType } from './constants'

export const gameTiles: GameTile[] = [
    {
        id: TileType.Ground,
        name: 'TILE_INFO.GROUND.NAME',
        image: './assets/images/tiles/grass.jpg',
        descriptions: [
            'TILE_INFO.GROUND.DESC1',
            'TILE_INFO.GROUND.DESC2',
            'TILE_INFO.GROUND.DESC3',
        ],
    },
    {
        id: TileType.Ice,
        name: 'TILE_INFO.ICE.NAME',
        image: './assets/images/tiles/ice.jpg',
        descriptions: [
            'TILE_INFO.ICE.DESC1',
            'TILE_INFO.ICE.DESC2',
            'TILE_INFO.ICE.DESC3',
        ],
    },
    {
        id: TileType.Water,
        name: 'TILE_INFO.WATER.NAME',
        image: './assets/images/tiles/water.jpg',
        descriptions: [
            'TILE_INFO.WATER.DESC1',
            'TILE_INFO.WATER.DESC2',
        ],
    },
    {
        id: TileType.Wall,
        name: 'TILE_INFO.WALL.NAME',
        image: './assets/images/tiles/wall.jpg',
        descriptions: [
            'TILE_INFO.WALL.DESC1',
            'TILE_INFO.WALL.DESC2',
            'TILE_INFO.WALL.DESC3',
        ],
    },
    {
        id: TileType.ClosedDoor,
        name: 'TILE_INFO.CLOSED_DOOR.NAME',
        image: './assets/images/tiles/closed-door.jpg',
        descriptions: [
            'TILE_INFO.CLOSED_DOOR.DESC1',
            'TILE_INFO.CLOSED_DOOR.DESC2',
        ],
    },
    {
        id: TileType.OpenDoor,
        name: 'TILE_INFO.OPEN_DOOR.NAME',
        image: './assets/images/tiles/open-door.jpg',
        descriptions: [
            'TILE_INFO.OPEN_DOOR.DESC1',
            'TILE_INFO.OPEN_DOOR.DESC2',
        ],
    },
]
